modules/account-lib/src/coin/algo/utils.ts

```diff
@@ -0,0 +1,611 @@
+   * @returns {boolean} true if the transaction can be decoded, otherwise false
+   */
+  protected isDecodableUnsignedAlgoTxn(txn: Uint8Array): boolean {
+    try {
+      algosdk.decodeUnsignedTransaction(txn);
+      return true;
+    } catch (_: unknown) {
+      return false;
+    }
+  }
+
+  /**
+   * Checks if a signed algo transaction can be decoded.
+   *
+   * @param {Uint8Array} txn The encoded signed transaction.
+   * @returns {boolean} true if the transaction can be decoded, otherwise false
+   */
+  protected isDecodableSignedTransaction(txn: Uint8Array): boolean {
+    try {
+      algosdk.decodeSignedTransaction(txn);
+      return true;
+    } catch (_: unknown) {
+      return false;
+    }
+  }
+
+  /**
+   * Decodes a signed or unsigned algo transaction.
+   *
+   * @param {Uint8Array | string} txnBytes The encoded unsigned or signed txn.
+   * @returns {EncodedTx} The decoded transaction.
+   */
+  decodeAlgoTxn(txnBytes: Uint8Array | string): EncodedTx {
+    let buffer;
+    if (typeof txnBytes === 'string') {
+      if (allHexChars(txnBytes)) {
+        buffer = Buffer.from(txnBytes, 'hex');
+      } else {
+        buffer = Buffer.from(txnBytes, 'base64');
+      }
+    } else {
+      buffer = txnBytes;
+    }
+
+    if (this.isDecodableUnsignedAlgoTxn(buffer)) {
+      return {
+        rawTransaction: new Uint8Array(buffer),
+        txn: algosdk.decodeUnsignedTransaction(buffer),
+        signed: false,
+      };
+    } else if (this.isDecodableSignedTransaction(buffer)) {
+      // TODO: Replace with
+      // return algosdk.Transaction.from_obj_for_encoding(algosdk.decodeSignedTransaction(buffer).txn);
+      // see: https://github.com/algorand/js-algorand-sdk/issues/364
+      // "...some parts of the codebase treat the output of Transaction.from_obj_for_encoding as EncodedTransaction.
+      // They need to be fixed(or we at least need to make it so Transaction conforms to EncodedTransaction)."
+      const tx: any = algosdk.decodeSignedTransaction(buffer);
+
+      const signers: string[] = [];
+      const signedBy: string[] = [];
+      if (tx.msig && tx.msig.subsig) {
+        for (const sig of tx.msig.subsig) {
+          const addr = algosdk.encodeAddress(sig.pk);
+          signers.push(addr);
+          if (sig.s) {
+            signedBy.push(addr);
+          }
+        }
+      }
+
+      return {
+        rawTransaction: new Uint8Array(buffer),
+        txn: tx.txn,
+        signed: true,
+        signers: signers,
+        signedBy: signedBy,
+      };
+    } else {
+      throw new InvalidTransactionError('Transaction cannot be decoded');
+    }
+  }
+
+  /**
+   * Estimate the transaction byte size by signing it with a random hey and calculating the byte length
+   *
+   * @param txInfo {Object} required fields to build a multisig transaction
+   * @returns {number} estimated byte size of the signed transaction
+   */
+  getTransactionByteSize(txInfo): number {
```

Comment:
It's a algosdk transaction

Reply:
When I say "what is the param type", I'm actually asking you to add a type to the param, not just tell me in a comment. Let's create a ticket and come back to this please since this is part of the module boundary.